config/initializers/puppet.rb

```diff
@@ -12,7 +12,7 @@
     :priority => 100
 end
 
-FactImporter.register_fact_importer :puppet, PuppetFactImporter, true
+Foreman::Plugin::ImporterRegistry.register :puppet, 'PuppetFactImporter', true
```

Comment:
Can we leave the ability to register by actual constant (as opposed to string). It will prevent typos - we are relying on ruby that the constant actually exists this way.

I don't want to change the inner implementation (that stores strings), just the outer interface. We can convert the constant to string inside the `register` method.

Reply:
I think we should register by using a string, 2 reasons:
1) If we use the original constant then it will get loaded at the beginning, even when it is not called for.
2) In the registry, I have used `@importers[key].constantize` which will take care of constantizing the string.